lib/hexpm/repository/release.ex

```diff
@@ -155,6 +155,34 @@ defmodule Hexpm.Repository.Release do
       select: {p.name, r.version, r.inserted_at, p.meta}
     )
   end
+
+  def downloads_by(release_id, filter) do
+    case filter do
+      :by_day ->
+        from(d in Download,
+          where: d.release_id == ^release_id,
+          group_by: fragment("date_trunc('day', ?)", d.day),
+          order_by: fragment("date_trunc('day', ?)", d.day),
+          select: {fragment("to_char(date_trunc('day', ?), 'YYYY-MM-DD')", d.day), fragment("sum(?)", d.downloads)})
+      :by_week ->
+        from(d in Download,
+          where: d.release_id == ^release_id,
+          group_by: fragment("date_trunc('week', ?)",  d.day), # d.day,
+          order_by: fragment("date_trunc('week', ?)",  d.day), # d.day,
+          select: {fragment("to_char(date_trunc('week', ?), 'YYYY-MM-DD')", d.day), fragment("sum(?)", d.downloads)})
+      :by_month ->
+        from(d in Download,
+          where: d.release_id == ^release_id,
+          group_by: fragment("date_trunc('month', ?)",  d.day), # d.day,
+          order_by: fragment("date_trunc('month', ?)",  d.day), # d.day,
+          select: {fragment("to_char(date_trunc('month', ?), 'YYYY-MM')", d.day), fragment("sum(?)", d.downloads)})
+      _ ->
```

Comment:
It's better to be strict here otherwise we can break clients when we introduce new filters. Only accept day/week/month in the downloads parameter and default to the current behavior when no parameter is given.